### PR TITLE
Dev/tests

### PR DESCRIPTION
**Pull request by ggrondin**

### CODE REVIEW

File: cfdtools/_cli.py [hunk 1]
```diff
@@ -5,8 +5,6 @@
 
 # readers and writers - must be imported to update api format dict
 import cfdtools.ic3 as ic3  # .reader_legacy # needed to map readers
-import cfdtools.gmsh as gmsh
```
Comment (jgressier) [inline cfdtools/_cli.py:8]: I thought it was compulsory to add them in order to fill the dictionnary of readers, did you check it still works ? (for example, ic3write command from a cgns file)

File: cfdtools/_cli.py [hunk 2]
```diff
@@ -111,13 +105,13 @@ def parse_filenameformat(self):
...
 
 
+@cli_header
 def info(argv=None):
     """call specific printinfo function from reader
 
     Args:
         argv (_type_, optional): _description_. Defaults to None.
     """
-    cli_header("cfdinfo")
```
Comment (jgressier) [inline cfdtools/_cli.py:120]: there, the name of function is "info" and it will replace "cfdinfo"

File: cfdtools/cgns/cgns.py [hunk 3]
```diff
@@ -1,11 +1,18 @@
 # cgns.py
 from pathlib import Path
-from cfdtools.api import io, error_stop, fileformat_reader, memoize
+from cfdtools.api import memoize as cache  # python <= 2.7
```
Comment (jgressier) [inline cfdtools/cgns/cgns.py:3]: cfdtools needs at least python 3.6

**Comment by jgressier** [comment]
do not hesitate to add some general comments about your devs (dev to dev ;-) )

File: cfdtools/_cli.py (hunk 1 above)
Comment (ggrondin) [reply cfdtools/_cli.py:8]: I have tried to change `test_1_cgns.py` with:
```
+import cfdtools.ic3.writerV3 as ic3writer
+import cfdtools.api as api
...
+    ic3write = ic3writer.writer(rmesh)
+    _builddir.mkdir(exist_ok=True)
+    outfile = api._files(_builddir / filename)
+    outfile.change_suffix('.ic3')
+    ic3write.write_data(outfile.filename)
+    Path(outfile.filename).unlink()
```
And it worked. So I think this is ok ?

File: cfdtools/cgns/cgns.py (hunk 3 above)
Comment (ggrondin) [reply cfdtools/cgns/cgns.py:3]: Then no need for `memoize`and:
```
try:
    from functools import cache  # python >= 3.9
except ImportError:
    from functools import lru_cache  #  3.6 <= python <= 3.8
    cache = lru_cache(maxsize=None)
    del lru_cache
```

File: cfdtools/_cli.py [hunk 4]
```diff
@@ -195,23 +189,23 @@ def write_generic(argv, ext, writer):
...
 
+@cli_header
 def write_vtk(argv=None):
-    cli_header("cfdwrite_vtk")
     return write_generic(argv, '.vtu', vtk.vtkMesh)
 
 
+@cli_header
 def writecube(argv=None):
-    cli_header("cfdwritecube")
```
[the remark this answers is not in the document]
Comment (ggrondin) [reply cfdtools/_cli.py:214]: Also added an `altfname` (what I started with before noticing all were prefixes...):
```
# For use with a prefix:    @cli_header("...")
# For use with altfname:    @cli_header(altfname="...")
def cli_header(prefix=None, altfname=None):
    def namedecorator(func):
        def decorator(*args, **kwargs):
            fname = func.__name__ if altfname is None else altfname
            if prefix is not None:
                fname = prefix + fname
            api.io.print('std', f"CFDTOOLS - {fname}")
            return func(*args, **kwargs)
        return decorator
    return namedecorator
```

**Review by jgressier — APPROVED** [review]

**Pull request merged**